sqlglot/expressions.py

```diff
@@ -3167,9 +3168,9 @@ class Neg(Unary):
     pass
 
 
-# Special Functions
 class Alias(Expression):
-    arg_types = {"this": True, "alias": False}
+    # "reverse" controls the relative order of "this" and "alias" for SQL generation
+    arg_types = {"this": True, "alias": False, "reverse": False}
```

Comment:
I added "reverse" here, because the order is reversed in the DEFINE clause, see test case:

```sql
...
UP AS UP.units_sold > PREV(UP.units_sold),  -- alias goes first...
...
```

There are at least three different ways to make this work:

1) The way I did it in this PR.
2) Swap the values of "this" and "alias". Note that this will create a wrong / misleading AST node and is a hack.
3) Use the `meta` property instead, like `meta["generate_in_reverse"] = True`. This will require creating a helper method for parsing aliases in reverse, because we need to set the meta after the expression is created and so can't be done in the lambda (see code in parser).

This feels like a niche feature, so I'm not sure which approach is the best.